src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/TrainerGenerators.cs

```diff
@@ -556,5 +556,35 @@ public override string[] GenerateUsings()
                 return _binaryTrainerUsings;
             }
         }
+
+        internal class MatrixFactorization : TrainerGeneratorBase
+        {
+            //ClassName of the trainer
+            internal override string MethodName => "MatrixFactorization";
+
+            internal override string OptionsName => "MatrixFactorizationTrainer.Options";
+            protected override bool IncludeFeatureColumnName => false;
+
+            //The named parameters to the trainer.
+            internal override IDictionary<string, string> NamedParameters
+            {
+                get
+                {
+                    return
```

Comment:
This is a bit of an anti-pattern: to return a new object every time a property is invoked. I see this is already being used in existing code, so you don't need to fix it in this PR. But in general, calling a property is supposed to be fast.

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property)

> Although properties are technically very similar to methods, they are quite different in terms of their usage scenarios. They should be seen as smart fields.